drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -882,17 +909,19 @@ local function level_attr_handler(driver, device, ib, response)
 end
 
 local function hue_attr_handler(driver, device, ib, response)
-  if ib.data.value ~= nil then
-    local hue = math.floor((ib.data.value / 0xFE * 100) + 0.5)
-    device:emit_event_for_endpoint(ib.endpoint_id, capabilities.colorControl.hue(hue))
+  if ib.data.value == nil or ignore_initial_color_read(device, color_mode_attr_bits.HUE) then
```

Comment:
Maybe I'm overthinking, but I think it would be good to change the order of these statements in all cases: 
```suggestion
  if ignore_initial_color_read(device, color_mode_attr_bits.HUE) or ib.data.value == nil then
```
to ensure we _always_ ignore the first read and only the first read, irrelevant to whether we receive a nil value or not.